test_venv.sh

```diff
@@ -0,0 +1,50 @@
+#!/bin/bash
+
+set -e
+
+_CDA_USE_VENV=0
+if [ -z "${VENV+x}" -a -n "$VIRTUAL_ENV" ] ; then
+    printf '%s\n' "INFO: Use existing VENV=\"$VIRTUAL_ENV\""
+elif [ -z "${VENV-x}" ]; then
+    printf '%s\n' "INFO: Don't use VENV"
+else
+    _CDA_USE_VENV=1
+    if [ -z "$VENV" ] ; then
+        VENV=ocp-venv
+    fi
+    printf '%s\n' "INFO: Use \"$VENV\" virtual environment"
+    if [ ! -f "$VENV/bin/activate" ] ; then
+        if [ -z "$VENV_PYTHON" ] ; then
+            _CDA_VENV_PYTHON=python
+        else
+            _CDA_VENV_PYTHON="$VENV_PYTHON"
+        fi
+        "$_CDA_VENV_PYTHON" -m venv "$VENV"
+    fi
+    source "$VENV/bin/activate"
+fi
+
+printf '%s\n' "INFO: python=$(printf '%q' "$(command -v python)")"
+
+python -m ensurepip --upgrade
+python -m pip install --upgrade pip
+cat ./requirements.txt  | xargs -n1 python -m pip install --upgrade
+
+EXTRA=(
```

Comment:
should be part of the requirements.txt file? Or are you saying the things below are only for linting and should be installed on your dev env  (not needed for CDA). If so, I would move these things into your own dev env.